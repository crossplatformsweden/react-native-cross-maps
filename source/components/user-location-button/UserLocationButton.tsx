import React from 'react';
import { View } from 'react-native';
import { Colors, CrossButton, styles } from 'react-native-cross-components';

interface IProps {
  onPress: () => void;
}

export const UserLocationButton = ({ onPress }: IProps) => (
  <View style={[styles.absoluteTopRight, { top: 20, right: 0 }]}>
    <CrossButton
      size={40}
      onPress={onPress}
      iconName="dot-circle-o"
      color={Colors.CrossBlack}
      backgroundColor="transparent"
    />
  </View>
);

export default UserLocationButton;
