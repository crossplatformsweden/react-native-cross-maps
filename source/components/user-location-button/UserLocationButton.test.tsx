import React from 'react';
import TestRenderer from 'react-test-renderer';
import { UserLocationButton } from './UserLocationButton';
import { CrossButton } from 'react-native-cross-components';

jest.unmock('react-native');

const wrapper = TestRenderer.create(
  <UserLocationButton onPress={() => console.log('pressed')} />
);

describe('components', () => {
  describe('UserLocationButton', () => {
    it('should render self and subcomponents', () => {
      const json = wrapper.toJSON();
      expect(json).toMatchSnapshot();
    });

    it('button chould have child button', () => {
      const btn = wrapper.root.findByType(CrossButton);
      expect(btn.props.iconName).toBe('dot-circle-o');
    });
  });
});
