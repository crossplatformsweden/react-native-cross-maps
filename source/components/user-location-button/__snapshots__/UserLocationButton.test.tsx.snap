// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components UserLocationButton should render self and subcomponents 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "right": 0,
        "top": 20,
      },
    ]
  }
>
  <Button
    backgroundColor="transparent"
    color="#2f2f2f"
    iconName="dot-circle-o"
    onPress={[Function]}
    size={40}
  />
</View>
`;
